posts/2023-12-30_containers_webassembly_submission/containers_and_webassembly_submissions.qmd

```diff
@@ -0,0 +1,255 @@
+
+#### Containers
+
+![](images/podman-docker-singularity.png){fig-align="center"}
+
+Containerization, particularly through technologies like [Docker, Podman or Singularity](https://appsilon.com/docker-vs-podman-vs-singularity/amp/), offers several advantages for deploying Shiny apps.
+
+##### Choosing the Right Container
+
+Choosing the right container was a question that arose in this project. Although **Docker** is the most popular, we decided to move forward with **Podman**. 
+
+In our exploration of containerization tools for deploying Shiny applications, we've identified key distinctions between Docker and Podman that influenced our choice. 
+
+**Podman** stands out for its daemonless architecture, enhancing security by eliminating the need for a central daemon process. Unlike Docker, Podman supports running containers as non-root users, a critical feature for meeting FDA reviewer requirements. Developed by Red Hat and maintained as an open-source project, Podman prioritizes security with its rootless container support, offering a robust solution for security-conscious users. 
+
+##### Goals
+
+A Container-based method to deploy Pilot 2 Shiny App.
+
+##### What we did
+
+1.  Configurable Podman Dockerfile / docker-compose.yml
+
+    a.  R version
+    b.  Registry / organization name / image name (differences between docker.io and ghcr.io)
+
+2.  Documentation on creating the container
+
+3.  CI: Automated build on amd64 and arm64 platforms
+
+##### Podman short-demo
+
+{{< video https://www.youtube.com/watch?v=H2WnpvVgmyE >}}
+
+Below is the dockerfile (recipe) for the container:
+
+![](images/file-recipe.png){fig-align="center"}
+
+# Next Steps
+
+The next steps are waiting for the review of the Pilot 3 by the FDA and to submit the two sections that explore the new technologies to regulatory authorities. Thanks to the collaboration between the R Submission Working Group and other institutions, there is already a working prototype of a {teal}-like Shiny application running on webR and further exploration with Podman is underway. 
+
+1.  Submission to FDA
+2.  Rhino Compatibility
+
+![](images/rhino.png){fig-align="center" width="161"}
+
+Appsilon is working on the side with Rhino compatibility, eventually we might be able to just add this framework into the Pilot 4 application.
+
+3.  {teal}
+
+    This and other packages might be on CRAN soon. We could incorporate them after that, replacing the shims created for this version
+
+4.  Boot Time
+
+    We need to improve boot time (remove dependencies and keep working on webR).
+
+<!--------------- appendices go here ----------------->
+
+source("appendix.R") insert_appendix( repo_spec = "pharmaverse/blog", name = long_slug ) \`\`\`
```

Comment:
can this get wrapped in R code chunk or removed?  We are still unsure what we are doing with this section.